store/cache/doublecache.go

```diff
@@ -29,6 +30,20 @@ func (mc *DoubleCache) Get(height uint64) (eds.AccessorStreamer, error) {
 	return mc.second.Get(height)
 }
 
+// GetOrLoad attempts to get an item from the both caches and, if not found, invokes
+// the provided loader function to load it into the first Cache.
+func (mc *DoubleCache) GetOrLoad(
+	ctx context.Context,
+	height uint64,
+	loader OpenAccessorFn,
+) (eds.AccessorStreamer, error) {
+	accessor, err := mc.second.Get(height)
```

Comment:
In `Get` we access `first` and then `second` cache (which sounds right). Why in `GetOrLoad` we accessing `second` and then `first` ?

Reply:
Read is optimistic (no tx gurantees). Order does not matter for read. `GetOrLoad` should load into first cache, so it checks `second` first. I'll add comments do clarify

Reply:
I didn't mention tx at all. But order indeed looks weird. Comment should be enough, thanks.

Reply:
Added comments